### PR TITLE
Provider agnostic forecast

### PR DESCRIPTION
**Pull request by luke-engle**
## What's changing?
Adding a provider-agnostic forecast command to go along with https://github.com/github/valet/pull/4353

## How's this tested?
```bash
➜ dotnet run --project src/Valet/Valet.csproj -- forecast -h                                        
Options:
  --source-file-path <source-file-path> (REQUIRED)  The file path(s) to existing jobs data.
  -o, --output-dir <output-dir> (REQUIRED)          The location for any output files.
  --start-date <start-date>                         The start date of the forecast analysis in YYYY-MM-DD format. [default: 6/15/2022 2:04:24 PM]
  --time-slice <time-slice>                         The time slice in seconds to use for computing concurrency metrics. [default: 60]
  --credentials-file <credentials-file>             The file containing the credentials to use.
  --no-telemetry                                    Boolean value to disallow telemetry.
  --no-ssl-verify                                   Disable ssl certificate verification.
  --no-http-cache                                   Disable caching of http responses.
  -?, -h, --help                                    Show help and usage information

Commands:
  azure-devops  Forecasts GitHub Actions usage from historical Azure DevOps pipeline utilization.
  jenkins       Forecasts GitHub Actions usage from historical Jenkins pipeline utilization.
  gitlab        Forecasts GitHub Actions usage from historical GitLab pipeline utilization.
  circle-ci     Forecasts GitHub Actions usage from historical CircleCI pipeline utilization.
  travis-ci     Forecasts GitHub Actions usage from historical Travis CI pipeline utilization.
```

```bash
➜ dotnet run --project src/Valet/Valet.csproj -- forecast jenkins -o tmp
[2022-06-22 18:59:53] Logs: 'tmp/log/valet-20220622-185953.log'
[2022-06-22 18:59:53] Forecasting 'https://dev.azure.com/valet-testing/build-playground/_build'
[2022-06-22 18:59:55] Output file(s):                                           
[2022-06-22 18:59:55]   tmp/jobs/06-22-2022-18-59_jobs_0.json
[2022-06-22 18:59:55]   tmp/forecast_report.md
➜ dotnet run --project src/Valet/Valet.csproj -- forecast azure-devops -o tmp
[2022-06-22 19:00:08] Logs: 'tmp/log/valet-20220622-190008.log'
[2022-06-22 19:00:08] Forecasting 'https://jenkout.westus2.cloudapp.azure.com'
[2022-06-22 19:01:27] Output file(s):                                           
[2022-06-22 19:01:27]   tmp/jobs/06-22-2022-19-01_jobs_0.json
[2022-06-22 19:01:27]   tmp/forecast_report.md
➜ dotnet run --project src/Valet/Valet.csproj -- forecast --source-file-path tmp/jobs/**.json -o tmp
[2022-06-22 19:10:19] Logs: 'tmp/log/valet-20220622-191019.log'
[2022-06-22 19:10:19] Forecasting '/data/tmp/jobs/06-22-2022-18-59_jobs_0.json, /data/tmp/jobs/06-22-2022-19-01_jobs_0.json'
[2022-06-22 19:10:19] Output file(s):                                           
[2022-06-22 19:10:19]   tmp/forecast_report.md
```

Closes #4364

### CODE REVIEW

File: src/Valet/Commands/Forecast.cs
Comment (lineville) [inline src/Valet/Commands/Forecast.cs:47]: What's the difference between adding as a global option, and doing it this way?

**Review by lineville — APPROVED** [review]
Looks good 🚀 ! Was the `--source-file-path` option not working in gh-valet prior to this?

File: src/Valet/Commands/Forecast.cs
Comment (luke-engle) [reply src/Valet/Commands/Forecast.cs:47]: Defining it as a global option would make `--source-file-path` a required option for all subcommands. The way it's written now, `--source-file-path` is required for the base forecast command (provider-agnostic), and optional when forecast is called with a subcommand (`forecast jenkins ...`)

**Comment by luke-engle** [comment]
> Looks good 🚀 ! Was the `--source-file-path` option not working in gh-valet prior to this?

I believe it was... mostly 😅  Though the help text for `--source-file-path` was certainly incorrect since most of the providers were using the one defined for the other commands

**Review by j-dunham — APPROVED** [review]

**Review by ethanis — APPROVED** [review]
LGTM.

Should this hold off on merging until github/valet#4353 is merged?

**Comment by luke-engle** [comment]
> LGTM.
> 
> Should this hold off on merging until [github/valet#4353](https://github.com/github/valet/pull/4353) is merged?

Yes, but it just merged 🙂

**Pull request merged**
